res/controllers/common-controller-scripts.js

```diff
@@ -128,6 +140,56 @@ var colorCodeToObject = function(colorCode) {
 var script = function() {
 };
 
+/**
+ * Discriminates whether an object was created using the `{}` synthax.
+ *
+ * Returns true when was an object was created using the `{}` synthax.
+ * False if the object is an instance of a class like Date or Proxy or an Array.
+ *
+ * isSimpleObject({}) // true
+ * isSimpleObject(null) // false
+ * isSimpleObject(undefined) // false
+ * isSimpleObject(new Date) // false
+ * isSimpleObject(new (class {})()) // false
+ * @param {any} obj Object to test
+ * @returns {boolean} true if obj was created using the `{}` or `new Object()` synthax, false otherwise
+ */
+const isSimpleObject = function(obj) {
+    return obj !== null && typeof obj === "object" && obj.constructor.name === "Object";
+};
```

Comment:
Is there a reason, why this is not part of `script`?

Reply:
No. Not really.

Reply:
#11625 make script a frozen object. Would like to merge this PR first and than unify it in #11625 .

Reply:
Ok, I amended the commit to add it to `script`.